### ISSUE TITLE
Requirement for encode_base64url() in MIME::Base64 implies version 3.11

### ISSUE THREAD

**Issue by wyoung** [PerlDancer/Dancer2#1086]
This function wasn't added to MIME::Base64 until version 3.11, but the version requirement isn't explicit in Dancer2. On trying to run Dancer2 on my CentOS 5 system, which ships MIME::Base64 version 3.07, you get an obscure "internal server error" complaint in the browser when trying to view even the sample Dancer2 app. ("dancer2 -a app")

**Comment by xsawyerx** [comment]
:+1: Nice discovery!

**Comment by veryrusty** [comment]
Our `dist.ini` has contained a minimum version for MIME::Base64 since Feb 2013:
```
[Prereqs]
...
MIME::Base64 = 3.13
```

@wyoung how did you install Dancer2 on that CentOS5 system ?

**Comment by wyoung** [comment]
@veryrusty: I believe I used cpanm, and it couldn't have been any earlier than December 2013, when I first started evaluating Dancer. I don't have access to the machine to check versions right now, since I'm home for the long weekend, but it was probably more recently than that.

I have a CentOS 5 test VM here, and I installed Dancer2 via cpanm on it. It replaced the platform-default version of MIME::Base64 with 3.15, the current version.

Because the platform ships with an older version, though, this means the problem could be that a later Perl RPM package update would replace the cpanm-provided version. Therefore, I recommend that you reinforce the version requirements in your dist.ini in the `use` statements as well:

    use MIME::Base64 v3.110;

I realize that this is a violation of the DRY principle, but it is a necessary guard.
